sphinx/config.py

```diff
@@ -31,6 +31,11 @@
                       "defaults to `{default.__name__}.'"
 
 
+string_classes = [text_type]
```

Comment:
`six.string_types` equals `basestring`, not `str`.
In `Config.check_types()`, it will be compared by `is` operator. So it is required the both `str` and `unicode`.